web/pages/Character/ChatList.tsx

```diff
@@ -41,32 +41,68 @@ const sortOptions = [
   { value: 'character-created', label: 'Bot Created', kind: 'bot' },
 ]
 
+type ChatCharacter = { _id: string; name: string; description: string; avatar: string | undefined }
+
+type ChatLine = {
+  _id: string
+  characters: ChatCharacter[]
+  name: string
+  createdAt: string
+  updatedAt: string
+}
+
 const CharacterChats: Component = () => {
   const params = useParams()
   const cache = getListCache()
-  const state = chatStore((s) => ({ list: s.all?.chats || [], chars: s.all?.chars || {} }))
+  const state = chatStore(
```

Comment:
Function instantiation in JS is fairly expensive. Considerably more expensive than it is in C#.
If it were `1` it'd be fine, but this is `N*M` which can be costly.